typescript/infra/scripts/funding/fund-keys-from-deployer.ts

```diff
@@ -418,10 +417,14 @@ class ContextFunder {
         const failure = await this.attemptToFundKey(key, chain);
         failureOccurred ||= failure;
       }
+      return failureOccurred;
     });
 
+    let failureOccurred = false;
     try {
-      await Promise.all(promises);
+      failureOccurred = (await Promise.all(promises)).some(
+        (promiseFailure) => promiseFailure,
+      );
     } catch (e) {
```

Comment:
this catch doesn't work for all the failureOccurred, only the ones caught in "Promise.all(promises)).some". Failures on line 403 or 418 just get returned as boolean. Can we move all of this into a fallible functions into try catch so we can catch there here?
Otherwise, the "'At least one failure occurred when funding'" message on line 287 isn't super helpful esp with multiple contexts.